docs/reference/style-guide.rst

```diff
@@ -0,0 +1,392 @@
+
+            Heading
+            .......   
+            
+
+    .. grid-item::
+        :child-align: center
+
+        ``<h5>`` heading
+
+Underlines must be the same length as the title or heading.
+
+Adhere to the following conventions:
+
+- Do not use consecutive headings without intervening text.
+- Be consistent with the characters you use for each level. Use the ones specified above.
+- Do not skip levels (for example, do not follow an H2 heading with an H4 heading).
+- Use sentence style for headings (capitalise only the first word).
+
+Inline formatting
+-----------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input
+     - Output
+   * - ``*Italic*``
+     - *Italic*
+   * - ``**Bold**``
+     - **Bold**
+   * - ````code````
+     - ``code``
+   * - ``:term:`Ubuntu```
+     - :term:`Ubuntu`
+   * - ``:file:`/file/path```
+     - :file:`/file/path`
+   * - ``:manpage:`man(1)```
+     - :manpage:`man(1)`
+   * - ``:command:`command```
+     - :command:`command`
+   * - ``:kbd:`key```
+     - :kbd:`key`
+
+Adhere to the following conventions:
+
+- Use italics sparingly. Common uses for italics are titles and names (for example,
+  when referring to a section title that you cannot link to, or when introducing
+  the name for a concept).
+- Use bold sparingly. Avoid using bold for emphasis and rather rewrite the sentence
+  to get your point across.
+
+
+Notes
+-----
+
+.. grid:: 2
+    :gutter: 2
+    
+    .. grid-item::
+        :child-align: center
+
+        .. code:: reStructuredText
+
+            .. note::
+                A note.
+
+    .. grid-item-card::
+
+            .. note::
+                A note.
+
+    .. grid-item::
+        :child-align: center
+
+        .. code:: reStructuredText
+
+            .. tip::
+                A tip.
+
+    .. grid-item-card::
+
+            .. tip::
+                A tip. 
+    
+    .. grid-item::
+        :child-align: center
+
+        .. code:: reStructuredText
+
+            .. important::
+                Important information.
+
+    .. grid-item-card::
+
+            .. important::
+                Important information. 
+    
+    .. grid-item::
+        :child-align: center
+
+        .. code:: reStructuredText
+
+            .. caution::
+                This might damage your hardware!
+
+    .. grid-item-card::
+
+            .. caution::
+                This might damage your hardware!
+
+Adhere to the following conventions:
+
+- Use notes sparingly.
+- Only use the following note types: ``note``, ``tip``, ``important``, ``caution``
+- Only use a caution if there is a clear hazard of hardware damage or data loss.
+
+Code blocks
+-----------
+
+To start a code block, start a code block with ``.. code::``.
+The code block must be surrounded by empty lines.
+
+When starting a code block, you can specify the code language to enforce a specific
+sytax highlighting, but in many cases, the default sytax highlighting works just fine.
+
```

Comment:
There should be a mention here that the contents of the code block are determined by the indentation level. Anything that's indented will be included in the code block, even if there are empty lines between the content.